src/Functions/array/arrayFilter.cpp

```diff
@@ -49,7 +49,30 @@ ColumnPtr ArrayFilterImpl::execute(const ColumnArray & array, ColumnPtr mapped)
 
 REGISTER_FUNCTION(ArrayFilter)
 {
-    factory.registerFunction<FunctionArrayFilter>();
+    FunctionDocumentation::Description description = "Returns an array containing only the elements in `arrN` for which `func(arr1[i], ..., arrN[i])` return something other than `0`.";
```

Comment:
Looking at the example in l. 63-67, the function really returns only the elements from the first array argument `arr1`. We should change `arrN` to `arr`.

Reply:
@rschu1ze I find the current descriptions like this a little unclear, which is why I introduced this idea of 'source' and 'condition' arrays.

What do you think of 

`Returns an array containing only the elements in the source array for which a lambda function returns something other than `0`.`

Then syntax: `arrayFilter(func(x [, y1, ..., yN]), source [, cond1, ... , condN])]`

- `func(x [, y1, ..., yN])` - A lambda function `func(x [, y1, y2, ... yN]) → F(x [, y1, y2, ... yN])` which operates on elements of the source array (`x`) and condition arrays (`y`).
- `source` - The source array to process
- `[, cond1, ... , condN]` - Optional. N condition arrays providing additional arguments to the lambda function.

Reply:
Sure, your proposal make sense.